src/platform/forms-system/src/js/utilities/ui/index.js

```diff
@@ -115,7 +115,22 @@ export function scrollToFirstError() {
     // Don't animate the scrolling if there is an open modal on the page. This
     // prevents the page behind the modal from scrolling if there is an error in
     // modal's form.
-    if (!document.body.classList.contains('modal-open')) {
+
+    // We have to search the shadow root of web components that have a slotted va-modal
+    const isShadowRootModalOpen = Array.from(
+      document.querySelectorAll('va-omb-info'),
+    ).some(ombInfo =>
+      ombInfo.shadowRoot.querySelector(
+        'va-modal[visible]:not([visible="false"])',
+      ),
+    );
+
+    const isModalOpen =
+      document.body.classList.contains('modal-open') ||
+      document.querySelectorAll('va-modal[visible]:not([visible="false"])') ||
+      isShadowRootModalOpen;
+
+    if (!isModalOpen) {
```

Comment:
Have you tested that the `isModalOpen` condition will be truthy if a `<va-modal>` is open and the `<body>` doesn't have the `modal-open` class?

Reply:
@bkjohnson I've updated the query selectors and tested that they work as expected on vets-website. I did not test them in the context of the `scrollToFirstError` function.

Reply:
I didn't run the full function but I did a quick test with a `<va-modal>` on a page and calculated `isModalOpen` in the console and it gave us `true` like we wanted.

```diff
diff --git a/src/applications/coronavirus-vaccination-expansion/containers/IntroductionPage.jsx b/src/applications/coronavirus-vaccination-expansion/containers/IntroductionPage.jsx
index 365c6650fb..5ff80a3c21 100644
--- a/src/applications/coronavirus-vaccination-expansion/containers/IntroductionPage.jsx
+++ b/src/applications/coronavirus-vaccination-expansion/containers/IntroductionPage.jsx
@@ -66,6 +66,7 @@ class IntroductionPage extends React.Component {
   render() {
     return (
       <div className="schemaform-intro">
+        <va-modal visible>Testing</va-modal>
         <br />
         <FormTitle title="Sign up to get a COVID-19 vaccine at VA" />
         <span>
```


![Screenshot of a local page with a modal open and the console open in the browser dev tools. The JS used to determine if a modal was open which uses a `va-modal[visible]` query selector is pasted into the console and the output shows that the boolean is true.](https://user-images.githubusercontent.com/2008881/187306021-8c9ac8ba-b6d0-4dcc-936e-c9d17a4fbab0.png)

I'll approve the other PR so we can make a release and bump this version again.